### ISSUE TITLE
Jest: a transform must export a `process` or `processAsync` function.

### ISSUE THREAD

**Issue by iameduardod** [nrwl/nx#6659]
## Current Behavior
nx run web-angular-project:test
Error: Jest: a transform must export a `process` or `processAsync` function.

## Expected Behavior
Test runs 

## Steps to Reproduce
package.json

{
  "scripts": {
    "nx": "nx",
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "12.2.0",
    "@angular/cdk": "^12.1.4",
    "@angular/common": "12.2.0",
    "@angular/compiler": "12.2.0",
    "@angular/core": "12.2.0",
    "@angular/forms": "12.2.0",
    "@angular/material": "12.2.0",
    "@angular/platform-browser": "12.2.0",
    "@angular/platform-browser-dynamic": "12.2.0",
    "@angular/router": "12.2.0",
    "@nativescript/angular": "^12.0.0",
    "@nativescript/core": "~8.0.0",
    "@ngx-translate/core": "~13.0.0",
    "@ngx-translate/http-loader": "~6.0.0",
    "core-js": "^3.6.5",
    "dayjs": "^1.10.6",
    "document-register-element": "1.13.1",
    "nativescript-ngx-fonticon": "~7.0.0",
    "nativescript-theme-core": "^1.0.4",
    "regenerator-runtime": "0.13.7",
    "rxjs": "^6.6.0",
    "tslib": "^2.0.0",
    "undefined": "^0.1.0",
    "zone.js": "0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "12.1.4",
    "@angular-eslint/eslint-plugin": "12.3.1",
    "@angular-eslint/eslint-plugin-template": "12.3.1",
    "@angular-eslint/template-parser": "12.3.1",
    "@angular/compiler-cli": "12.2.0",
    "@angular/language-service": "12.2.0",
    "@nativescript/eslint-plugin": "~0.0.4",
    "@nativescript/types": "~8.0.0",
    "@nativescript/webpack": "beta",
    "@ngtools/webpack": "^12.0.0",
    "@nrwl/angular": "12.6.5",
    "@nrwl/cli": "12.6.5",
    "@nrwl/cypress": "12.6.5",
    "@nrwl/eslint-plugin-nx": "12.6.5",
    "@nrwl/jest": "12.6.5",
    "@nrwl/linter": "12.6.5",
    "@nrwl/tao": "12.6.5",
    "@nrwl/web": "12.6.5",
    "@nrwl/workspace": "12.6.5",
    "@nstudio/angular": "12.5.1",
    "@nstudio/nativescript": "12.5.1",
    "@nstudio/nativescript-angular": "12.5.1",
    "@nstudio/web": "12.5.1",
    "@nstudio/web-angular": "12.5.1",
    "@nstudio/xplat": "^12.5.1",
    "@storybook/addon-a11y": "^6.3.6",
    "@storybook/addon-actions": "^6.3.6",
    "@storybook/addon-essentials": "^6.3.6",
    "@storybook/addon-links": "^6.3.6",
    "@storybook/builder-webpack5": "^6.3.6",
    "@storybook/manager-webpack5": "^6.3.6",
    "@storybook/web-components": "^6.3.6",
    "@types/jest": "26.0.8",
    "@types/node": "14.14.33",
    "@typescript-eslint/eslint-plugin": "4.28.5",
    "@typescript-eslint/parser": "4.28.5",
    "babel-loader": "^8.2.2",
    "codelyzer": "~6.0.0",
    "copy-webpack-plugin": "^9.0.0",
    "css-loader": "^5.2.6",
    "cypress": "^7.3.0",
    "del-webpack-plugin": "^2.1.0",
    "dotenv": "8.2.0",
    "eslint": "~7.22.0",
    "eslint-config-prettier": "8.1.0",
    "eslint-plugin-cypress": "^2.10.3",
    "extract-loader": "^5.1.0",
    "jest": "27.0.4",
    "jest-jasmine2": "~25.2.0",
    "jest-preset-angular": "8.4.0",
    "lit": "^2.0.0-rc.2",
    "lit-scss-loader": "^1.1.0",
    "mini-css-extract-plugin": "2.2.0",
    "prettier": "^2.3.1",
    "resolve-url-loader": "^4.0.0",
    "sass": "^1.30.0",
    "sass-loader": "^12.1.0",
    "storybook-addon-themes": "^6.1.0",
    "ts-jest": "27.0.4",
    "ts-loader": "^9.2.3",
    "ts-node": "~9.1.1",
    "typescript": "4.3.5",
    "webpack": "^5.41.1",
    "webpack-cli": "^4.7.2",
    "webpack-dev-server": "4.0.0-rc.0"
  }
}

**Comment by AgentEnder** [comment]
Hey! It looks like jest-preset-angular is a major version behind our specifications in your package.json. Does updating it to 9.0.4 fix this issue?

**Comment by AgentEnder** [comment]
For reference, here are the versions I am looking at: https://github.com/nstudio/xplat/blob/master/packages/angular/src/utils/versions.ts

**Comment by NathanWalker** [comment]
We'll update https://github.com/nstudio/xplat/blob/master/packages/angular/src/utils/versions.ts today to support 12.6.x and likely begin pinning the patch version as there had been some breaking changes over past couple weeks - we will need 12.6.6 out before we can bump it due to this:
https://github.com/nrwl/nx/issues/6606#issuecomment-894335817

**Comment by NathanWalker** [comment]
@iameduardod You can likely run this:
```
nx migrate latest
```
And then follow the directions about running the migrations. That should help migrate your jest.config - I would wait until 12.6.6 of Nx is released though before migrating - you can monitor released today/tomorrow for when that's available:
https://github.com/nrwl/nx/releases

**Comment by iameduardod** [comment]
@NathanWalker sweet i appreciate the update!

**Comment by AgentEnder** [comment]
I'm going to close this out, since the root cause has been identified. If you have any further questions, feel free to ask them and this can be reopened or you can open a new issue if its unrelated 😄

**Comment by NathanWalker** [comment]
@iameduardod another hint in order to get all the migrations from Nx is you can pin your workspace to older versions, for example:
```
"@nrwl/angular": "12.5.0",
"@nrwl/cli": "12.5.0",
"@nrwl/cypress": "12.5.0",
"@nrwl/eslint-plugin-nx": "12.5.0",
"@nrwl/jest": "12.5.0",
"@nrwl/linter": "12.5.0",
"@nrwl/tao": "12.5.0",
"@nrwl/web": "12.5.0",
"@nrwl/workspace": "12.5.0",
```
Then clean with those now set: `npm run clean` (or `yarn clean`) - or even `npm run clean.all` (or `yarn clean.all`) if have one of those.
Then you can proceed to migrate as it will pick up the migrations from there to latest:
```
nx migrate latest
```
Which will allow you to run them against your workspace to ensure you can get on the latest without skipping migrations you may need.

**Comment by iameduardod** [comment]
@NathanWalker 

Completed the steps, and still getting the exact error. 

Here's my package.json


"dependencies": {
    "@angular/animations": "12.2.1",
    "@angular/cdk": "^12.1.4",
    "@angular/common": "12.2.1",
    "@angular/compiler": "12.2.1",
    "@angular/core": "12.2.1",
    "@angular/forms": "12.2.1",
    "@angular/material": "12.2.1",
    "@angular/platform-browser": "12.2.1",
    "@angular/platform-browser-dynamic": "12.2.1",
    "@angular/router": "12.2.1",
    "@microsoft/applicationinsights-web": "^2.6.5",
    "@nativescript/angular": "^12.0.0",
    "@nativescript/core": "~8.0.0",
    "@ngx-translate/core": "~13.0.0",
    "@ngx-translate/http-loader": "~6.0.0",
    "classlist.js": "^1.1.20150312",
    "core-js": "^3.6.5",
    "custom-event-polyfill": "^1.0.7",
    "dayjs": "^1.10.6",
    "document-register-element": "1.13.1",
    "nativescript-ngx-fonticon": "~7.0.0",
    "nativescript-theme-core": "^1.0.4",
    "regenerator-runtime": "0.13.7",
    "rxjs": "^6.6.0",
    "tslib": "^2.0.0",
    "undefined": "^0.1.0",
    "web-animations-js": "^2.3.2",
    "whatwg-fetch": "^3.6.2",
    "zone.js": "0.11.4"
  },
  "devDependencies": {
    "@adobe/adobe-client-data-layer": "^2.0.2",
    "@angular-devkit/build-angular": "12.1.4",
    "@angular-eslint/eslint-plugin": "12.3.1",
    "@angular-eslint/eslint-plugin-template": "12.3.1",
    "@angular-eslint/template-parser": "12.3.1",
    "@angular/compiler-cli": "12.2.1",
    "@angular/language-service": "12.2.1",
    "@nativescript/eslint-plugin": "~0.0.4",
    "@nativescript/types": "~8.0.0",
    "@nativescript/webpack": "beta",
    "@ngtools/webpack": "^12.0.0",
    "@nrwl/angular": "12.7.2",
    "@nrwl/cli": "12.7.2",
    "@nrwl/cypress": "12.7.2",
    "@nrwl/eslint-plugin-nx": "12.7.2",
    "@nrwl/jest": "12.7.2",
    "@nrwl/linter": "12.7.2",
    "@nrwl/tao": "12.7.2",
    "@nrwl/web": "12.7.2",
    "@nrwl/workspace": "12.7.2",
    "@nstudio/angular": "12.5.1",
    "@nstudio/nativescript": "12.5.1",
    "@nstudio/nativescript-angular": "12.5.1",
    "@nstudio/web": "12.5.1",
    "@nstudio/web-angular": "12.5.1",
    "@nstudio/xplat": "12.7.0",
    "@storybook/addon-a11y": "^6.3.6",
    "@storybook/addon-actions": "^6.3.6",
    "@storybook/addon-essentials": "^6.3.6",
    "@storybook/addon-links": "^6.3.6",
    "@storybook/builder-webpack5": "^6.3.6",
    "@storybook/manager-webpack5": "^6.3.6",
    "@storybook/web-components": "^6.3.6",
    "@types/jest": "26.0.8",
    "@types/node": "14.14.33",
    "@typescript-eslint/eslint-plugin": "4.28.5",
    "@typescript-eslint/parser": "4.28.5",
    "babel-loader": "^8.2.2",
    "codelyzer": "~6.0.0",
    "copy-webpack-plugin": "^9.0.0",
    "css-loader": "^5.2.6",
    "cypress": "^7.3.0",
    "del-webpack-plugin": "^2.1.0",
    "dotenv": "10.0.0",
    "eslint": "~7.22.0",
    "eslint-config-prettier": "8.1.0",
    "eslint-plugin-cypress": "^2.10.3",
    "extract-loader": "^5.1.0",
    "jest": "27.0.3",
    "jest-jasmine2": "~25.2.0",
    "jest-preset-angular": "8.4.0",
    "lit": "^2.0.0-rc.2",
    "lit-scss-loader": "^1.1.0",
    "mini-css-extract-plugin": "2.2.0",
    "nx": "^12.7.1",
    "prettier": "^2.3.1",
    "resolve-url-loader": "^4.0.0",
    "sass": "^1.30.0",
    "sass-loader": "^12.1.0",
    "storybook-addon-themes": "^6.1.0",
    "ts-jest": "27.0.3",
    "ts-loader": "^9.2.3",
    "ts-node": "~9.1.1",
    "typescript": "4.3.5",
    "webpack": "^5.41.1",
    "webpack-cli": "^4.7.2",
    "webpack-dev-server": "4.0.0-rc.0"
  }

**Comment by NathanWalker** [comment]
@iameduardod can you share your apps jest.config as well as the one at root of your workspace?

**Comment by iameduardod** [comment]
@NathanWalker  ah i found the fix! both migrate options did not include the update to jest-preset-angular. After updating to 9.0.6, everything worked!

**Comment by NathanWalker** [comment]
Great to hear thanks for letting us know.
